### DISCUSSION TITLE
Need Help: incorrectly reverse proxy settings (forwarded_for_headers + trusted_proxies)

### DISCUSSION THREAD

**Discussion by Rimos911** [nextcloud/all-in-one#4508] [Questions] [answered]
Hello,

can someone tell me if my reverse proxy is set correctly. Here is an excerpt from my config.php and a screenshot of the settings on the Sysonolgy DSM reverse proxy.

Is there something missing here or is something set incorrectly for my setting? The settings are default (except for array 2 with the IP)

I don't see any errors but it doesn't work as it should! I couldn´t found anything in the documantation.

config.php:

  'forwarded_for_headers' =>
  array (
    0 => 'HTTP_X_FORWARDED',
    1 => 'HTTP_FORWARDED_FOR',
  ),

... and ...

  'trusted_proxies' =>
  array (
    0 => '127.0.0.1',
    1 => '::1',
    2 => '192.168.2.18', 
  ),

Message orange (german)
Es gibt einige Warnungen bei deiner Systemkonfiguration.

Die Konfiguration des Reverse-Proxy-Headers ist falsch oder du greifst über einen vertrauenswürdigen Proxy auf Nextcloud zu. Andernfalls stellt dies ein Sicherheitsproblem dar und kann es einem Angreifer ermöglichen, seine IP-Adresse so zu fälschen, dass sie für Nextcloud sichtbar ist

Thnaks in Advanced

![Screenshot 2024-04-06 135031](https://github.com/nextcloud/all-in-one/assets/70541883/5bd53f49-bc19-4806-a2c7-a45250d33cb1)
![Screenshot 2024-04-06 134546](https://github.com/nextcloud/all-in-one/assets/70541883/304615bf-5958-4e2a-b24e-01e3413fed45)

on the other hand, under brute force protection I get this error message. Could it be that this is related?

![Screenshot 2024-04-06 140051](https://github.com/nextcloud/all-in-one/assets/70541883/ea0bd3fc-3f3d-4034-b594-0fae9d47a51d)

**Comment by szaimen** [comment, marked as answer]
> Could it be that this is related?
> 
> ![Screenshot 2024-04-06 140051](https://private-user-images.githubusercontent.com/70541883/320202935-ea0bd3fc-3f3d-4034-b594-0fae9d47a51d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI1NTAwOTcsIm5iZiI6MTcxMjU0OTc5NywicGF0aCI6Ii83MDU0MTg4My8zMjAyMDI5MzUtZWEwYmQzZmMtM2YzZC00MDM0LWI1OTQtMGZhZTlkNDdhNTFkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA4VDA0MTYzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFmMWE1ZDcwYzBhMGMwZGRjMTU5ZDFmZjBiOWNhYjA3ZjVlMTk3ZjI5ZTA4OGYwZWExODc3NzFmODBlZTM5OWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7xXqSSm7tqEVKh9VhjPj-C56bVvHuMKw39fzch1AQxw)

This is weird since ::1 is already part of the config list...

Can you follow https://github.com/nextcloud/all-in-one/discussions/2045?

**Comment by Rimos911** [comment]
I know now why I got this error/warning. The Linux-System has enabled IPv6.

With this proceed I permanently disable IPv6 on the fundamental Linux Server (i my case Ubuntu 22.04)

1. nano /etc/default/grub
2. change from:
   GRUB_CMDLINE_LINUX_DEFAULT=""
   to:
   GRUB_CMDLINE_LINUX_DEFAULT="ipv6.disable=1"
3. sudo update-grub
4. reboot

**Reply by lastsamurai26** [reply]
Thx this Solved the Problem for me aswell 
Problem is Why now ?

**Reply by lastsamurai26** [reply]
but Now
Collabora doesn't work 
wsd-00007-00007 2024-04-11 10:13:30.003453 +0200 [ coolwsd ] WRN  #-1: Failed setsockopt TCP_NODELAY. Will not report further failures to set TCP_NODELAY: Bad file descriptor| net/Socket.hpp:214
wsd-00007-00007 2024-04-11 10:13:30.003497 +0200 [ coolwsd ] TRC  #-1: Created socket. Thread affinity set to 0x7f7883323880| net/Socket.hpp:384
wsd-00007-00007 2024-04-11 10:13:30.003502 +0200 [ coolwsd ] ERR  #-1: Failed setsockopt SO_REUSEADDR: Bad file descriptor (EBADF: Bad file descriptor)| net/Socket.cpp:852
wsd-00007-00007 2024-04-11 10:13:30.003513 +0200 [ coolwsd ] ERR  #-1: Failed set ipv6 socket to 0 (EBADF: Bad file descriptor)| net/Socket.cpp:883
wsd-00007-00007 2024-04-11 10:13:30.003521 +0200 [ coolwsd ] ERR  #-1: Failed to bind to: IPv6 port: 9980 (EBADF: Bad file descriptor)| net/Socket.cpp:890

**Reply by lastsamurai26** [reply]
ok if we disable ipv6 directly in the kernel than collabora doesn't work anymore 
https://github.com/CollaboraOnline/online/issues/3102

**Reply by Rimos911** [reply]
I'm glad it helped you too - at least partially.

I can't tell you exactly. I can only assume that my server (Ubuntu) had IPv6 was enabled (probably during installation) and NC has a problem with the configuration of my reverse proxy (running on the disk station). I don´t user IPv& in my LAN or for the Internet
